elide-async/src/test/java/com/yahoo/elide/async/service/AsyncQueryThreadTest.java

```diff
@@ -42,37 +40,57 @@ public void setupMocks() {
         asyncQueryDao = mock(DefaultAsyncQueryDAO.class);
+        resultStorageEngine = mock(DefaultResultStorageEngine.class);
     }
 
     @Test
-    public void testProcessQueryJsonApi() throws NoHttpResponseException, URISyntaxException {
+    public void testProcessQueryJsonApi() throws Exception {
         AsyncQuery queryObj = new AsyncQuery();
         ElideResponse response = new ElideResponse(200, "ResponseBody");
         String query = "/group?sort=commonName&fields%5Bgroup%5D=commonName,description";
         String id = "edc4a871-dff2-4054-804e-d80075cf827d";
         queryObj.setId(id);
         queryObj.setQuery(query);
         queryObj.setQueryType(QueryType.JSONAPI_V1_0);
+        queryObj.setResultType(ResultType.EMBEDDED);
+
         when(elide.get(anyString(), any(), any(), anyString(), any())).thenReturn(response);
-        AsyncQueryThread queryThread = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, "v1");
+        AsyncQueryThread queryThread = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, "v1",
+                resultStorageEngine);
         queryResultObj = queryThread.processQuery();
         assertEquals(queryResultObj.getResponseBody(), "ResponseBody");
         assertEquals(queryResultObj.getHttpStatus(), 200);
     }
 
     @Test
-    public void testProcessQueryGraphQl() throws NoHttpResponseException, URISyntaxException {
+    public void testProcessQueryGraphQl() throws Exception {
         AsyncQuery queryObj = new AsyncQuery();
         ElideResponse response = new ElideResponse(200, "ResponseBody");
         String query = "{\"query\":\"{ group { edges { node { name commonName description } } } }\",\"variables\":null}";
         String id = "edc4a871-dff2-4054-804e-d80075cf827d";
         queryObj.setId(id);
         queryObj.setQuery(query);
         queryObj.setQueryType(QueryType.GRAPHQL_V1_0);
+        queryObj.setResultType(ResultType.EMBEDDED);
+
         when(runner.run(eq(query), eq(user), any())).thenReturn(response);
-        AsyncQueryThread queryThread = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, "v1");
+        AsyncQueryThread queryThread = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, "v1",
+                resultStorageEngine);
         queryResultObj = queryThread.processQuery();
         assertEquals(queryResultObj.getResponseBody(), "ResponseBody");
         assertEquals(queryResultObj.getHttpStatus(), 200);
     }
+
+    @Test
+    public void testConvertJsonToCSV() throws Exception {
+
+        String csvStr = "[key][\"value\"]";
+        String jsonStr = "{\"key\":\"value\"}";
+
+        AsyncQuery queryObj = new AsyncQuery();
+        AsyncQueryThread queryThread = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, "v1",
+                resultStorageEngine);
+
+        assertEquals(queryThread.convertJsonToCSV(jsonStr), csvStr);
+    }
```

Comment:
We need additional test cases for the calculateRecordCounts method with positive and negative scenarios.